src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -429,18 +429,35 @@ public Response getVersion(@Context ContainerRequestContext crc,
            
             //If excludeFiles is null the default is to provide the files and because of this we need to check permissions. 
             boolean checkPerms = excludeFiles == null ? true : !excludeFiles;
+            
 
             Dataset dst = findDatasetOrDie(datasetId);
-            DatasetVersion dsv = getDatasetVersionOrDie(req, versionId, dst, uriInfo, headers, includeDeaccessioned, checkPerms);
-
-            if (dsv == null || dsv.getId() == null) {
+            DatasetVersion requestedDatasetVersion 
+                    = getDatasetVersionOrDie(req, versionId, dst, uriInfo, headers, includeDeaccessioned,
+                    checkPerms);
+            
+            DatasetVersion latestDatasetVersion = null;
+            boolean deaccesionedLookup = true;
+            //Check perms is false since we are never going to retrieve files, we are just getting the status of the version.
+            checkPerms = false;
+            if(versionId != DS_VERSION_LATEST){                
+                latestDatasetVersion = getDatasetVersionOrDie(req, DS_VERSION_LATEST, dst, uriInfo, headers,
+                        deaccesionedLookup,
+                        checkPerms);
```

Comment:
The checkPerms is only false in the case that the request skip the files, the default is true since the default behavior of the API is to bring the files for the user.